src/OrchardCore.Modules/OrchardCore.AuditTrail/Shapes/ContentAuditTrailEventShapesTableProvider.cs

```diff
@@ -0,0 +1,118 @@
+using Microsoft.Extensions.DependencyInjection;
+using Newtonsoft.Json.Linq;
+using OrchardCore.AuditTrail.Extensions;
+using OrchardCore.AuditTrail.Indexes;
+using OrchardCore.AuditTrail.Models;
+using OrchardCore.AuditTrail.Services.Models;
+using OrchardCore.ContentManagement;
+using OrchardCore.ContentManagement.Metadata;
+using OrchardCore.DisplayManagement.Descriptors;
+using System.Collections.Generic;
+using System.Linq;
+using YesSql;
+using IYesSqlSession = YesSql.ISession;
+
+namespace OrchardCore.AuditTrail.Shapes
+{
+    public class ContentAuditTrailEventShapesTableProvider : IShapeTableProvider
+    {
+        public void Discover(ShapeTableBuilder builder)
+        {
+            builder.Describe("AuditTrailEvent").OnDisplaying(async context =>
+            {
+                dynamic shape = context.Shape;
+
+                var auditTrailEvent = (AuditTrailEvent)shape.AuditTrailEvent;
+
+                if (auditTrailEvent.Category != "Content") return;
+
+                var contentItemJToken = auditTrailEvent.Get(auditTrailEvent.EventName);
+                var contentEventJToken = auditTrailEvent.Get(nameof(ContentEvent));
+                var contentItem = contentItemJToken.ToObject<ContentItem>();
+
+                if (context.Shape.Metadata.DisplayType == "Detail")
+                {
+                    var session = context.ServiceProvider.GetRequiredService<IYesSqlSession>();
+
+                    AuditTrailEvent previousAuditTrailEvent = null;
+
+                    previousAuditTrailEvent = await session.Query<AuditTrailEvent, AuditTrailEventIndex>()
```

Comment:
Drivers. Too much to be using `IShapeTableProvider` for this.

Breaking changes in this module to fix it later